### DISCUSSION TITLE
Logs server doesn't fall back to local logs when Amazon CloudWatch logs aren't found

### DISCUSSION THREAD

**Discussion by matthewblock** [apache/airflow#27395] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

We recently activated AWS Cloudwatch logs. We were hoping the logs server would gracefully handle task logs that previously existed but were not written to Cloudwatch, but when fetching the remote logs failed (expected), the logs server didn't fall back to local logs.

```
*** Reading remote log from Cloudwatch log_group: <our log group> log_stream: <our log stream>
```

### What you think should happen instead

According to documentation [Logging for Tasks](https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/logging-tasks.html#writing-logs-locally), when fetching remote logs fails, the logs server should fall back to looking for local logs:

> In the Airflow UI, remote logs take precedence over local logs when remote logging is enabled. If remote logs can not be found or accessed, local logs will be displayed.

This should be indicated by the message `*** Falling back to local log`.

If this is not the intended behavior, the documentation should be modified to reflect the intended behavior.

### How to reproduce

1. Run a test DAG without [AWS CloudWatch logging configured](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/logging/cloud-watch-task-handlers.html)
2. Configure AWS CloudWatch remote logging and re-run a test DAG

### Operating System

Debian buster-slim

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Converted to discussion by mistake. This is really CloudWatch task implementation issue and should be easy - you can even attempt it by lookin at how S3 implemeted it. Creaed https://github.com/apache/airflow/issues/27396 and marked as good first issue. If you will not attempt to  fix it  @matthewblock, maybe someone else will pick it up.
